pallets/stateful-storage/src/tests/signed_extension_tests.rs

```diff
@@ -0,0 +1,191 @@
+use crate::{
+	signed_extension::StatefulStorageSignedExtension,
+	test_common::{constants::*, test_utility::*},
+	tests::mock::*,
+	types::*,
+	Call,
+};
+use common_primitives::stateful_storage::PageHash;
+use frame_support::{assert_ok, dispatch::DispatchInfo, BoundedVec};
+#[allow(unused_imports)]
+use pretty_assertions::{assert_eq, assert_ne, assert_str_eq};
+use sp_core::Pair;
+use sp_runtime::{
+	traits::SignedExtension,
+	transaction_validity::{InvalidTransaction, TransactionValidity, TransactionValidityError},
+	MultiSignature,
+};
+
+#[test]
+fn signed_extension_validation_with_stale_hash_should_fail() {
+	new_test_ext().execute_with(|| {
+		// arrange
+		let caller_1 = test_public(1);
+		let itemized_schema_id = ITEMIZED_SCHEMA;
+		let paginated_schema_id = PAGINATED_SCHEMA;
+		let payload = vec![1; 5];
+		let prev_content_hash: PageHash = 0;
+		let actions = vec![ItemAction::Add { data: payload.clone().try_into().unwrap() }];
+		let info = DispatchInfo::default();
+		let len = 0_usize;
+		let sig = [1; 64];
+		let signature = MultiSignature::Sr25519(sp_core::sr25519::Signature::from_raw(sig));
+		let (msa_id, delegator_1) = get_signature_account();
+		let delegator_key = delegator_1.public();
+		assert_ok!(StatefulStoragePallet::apply_item_actions(
+			RuntimeOrigin::signed(caller_1.clone()),
+			msa_id,
+			itemized_schema_id,
+			prev_content_hash,
+			BoundedVec::try_from(actions.clone()).unwrap(),
+		));
+		assert_ok!(StatefulStoragePallet::upsert_page(
+			RuntimeOrigin::signed(caller_1.clone()),
+			msa_id,
+			paginated_schema_id,
+			0,
+			prev_content_hash,
+			BoundedVec::try_from(payload.clone()).unwrap(),
+		));
+		let calls: Vec<<Test as frame_system::Config>::RuntimeCall> = vec![
+			RuntimeCall::StatefulStoragePallet(Call::apply_item_actions {
+				actions: BoundedVec::try_from(actions.clone()).unwrap(),
+				target_hash: NONEXISTENT_PAGE_HASH,
```

Comment:
maybe a nit, but I feel like using `NONEXISTENT_PAGE_HASH` (== 0) in the tests is a bit too simple; it feels like it has the potential to hit a special case in the code under test (I'm pretty sure it doesn't, but it feels weird to me anyway). Maybe better to use a random number? (Or test both cases: 0 and <random number>?)

Reply:
Changing to a random value on any run would create a flaky test with very low priority of failure but still using a deterministic stale value would be better. I'll use a different hash variable name to make it less confusing.